packages/core/src/jsx.js

```diff
@@ -112,6 +128,23 @@ export const jsx: typeof React.createElement = function(
     }
   }
   newProps[typePropName] = type
+  if (process.env.NODE_ENV !== 'production') {
+    try {
+      throw new Error()
```

Comment:
This is really clever 😄 Are there any performance implications?

Reply:
It's only for dev so it doesn't really matter much but I checked the benchmarks and it seems fine.